File: Eratosthenes.java
import java.util.ArrayList;

public class Eratosthenes{

	private int N=10000;
	private ArrayList<Integer> primeList;
	private ArrayList<Integer> removedList;

	public Eratosthenes(int N){
		this.N=N;
		primeList=new ArrayList<Integer>(N);
		removedList=new ArrayList<Integer>(N);

		for(int i=2;i<=this.N;i++)
			primeList.add(i);
	}
	
	public void Start(){

		primeList.removeIf(m -> (m % 2)==0);
		primeList.add(0,2);

		for(int i=3;i<=this.N;i+=2){
			
			if(removedList.indexOf(i)==-1){
				final int j=i;
				primeList.stream().filter(m -> m % j==0).forEach(x ->{
					removedList.add(x);
				});
				primeList.removeIf(m -> m % j==0);
				primeList.add(i);
				
			}
		}

		System.out.println(primeList);
		System.out.printf("Prime number sum : %d\n",primeList.size());
	}
}

File: Eratosthenes.cpp
#include "Eratosthenes.h"

nori::Eratosthenes::Eratosthenes(){
    this->N=10000;
}

nori::Eratosthenes::Eratosthenes(size_t N){
    this->N=N;
}

nori::Eratosthenes::~Eratosthenes(){
    delete this->numberList;
    delete this->removedList;
}

void nori::Eratosthenes::Start(){
    this->numberList=new std::vector<size_t>(N);
    this->removedList=new std::vector<size_t>(N);

    std::iota(this->numberList->begin(),this->numberList->end(),1);
    this->numberList->erase(std::remove_if(
            this->numberList->begin(),
            this->numberList->end(),
            [](size_t x){return x==1;}));

    this->numberList->erase(std::remove_if(
        this->numberList->begin(),
        this->numberList->end(),
        [](size_t x) { return x % 2 == 0; }),
        this->numberList->end());

    this->numberList->push_back(2);
    for(size_t i=3;i<=this->N;i+=2){
        //もしiが未探索の数字なら...
        if(std::find(this->removedList->begin(),this->removedList->end(),i)==this->removedList->end()){
            
            std::copy_if(this->numberList->begin(),this->numberList->end(),std::back_inserter(*this->removedList),[=](size_t x){return x % i==0;});
            
            this->numberList->erase(std::remove_if(
                this->numberList->begin(),
                this->numberList->end(),
                [=](size_t x){return x % i==0;}),
                this->numberList->end()
            );
            this->numberList->push_back(i);

        }
        
    }
    std::sort(this->numberList->begin(),this->numberList->end());
    for(auto& number : *this->numberList){
        std::cout << number << " ";   
    }
    std::cout << std::endl;
    std::cout << "prime Number sum : " << this->numberList->size() << std::endl;
}